backend/app/user_handler.go

```diff
@@ -350,6 +447,16 @@ func (h *Handler) LoginUserHandler(c *gin.Context) {
 		return
 	}
 
+	// refresh sponsorship status on login
+	if err := h.refreshUserVerification(&user); err != nil {
+		log.Error().Err(err).Msg("failed to refresh user sponsorship on login")
+	}
+
+	if !user.Sponsored {
```

Comment:
This is wrong. You won't reject login because the sponsored flag is false (which could also be coming from transient errors). you should allow the user to login and disable their ability to deploy (already handled by zos)